gtk/main.c

```diff
@@ -814,6 +814,10 @@ static gboolean winclose(GtkWidget* w UNUSED, GdkEvent* event UNUSED, gpointer g
     {
         gtr_action_activate("toggle-main-window");
     }
+    else if (gtr_pref_flag_get(TR_KEY_minimize_on_close))
```

Comment:
With the proposed UI changes, it's not clear to me (as a user) that minimizing will only work if system tray icon is turned off. Maybe we should disable the former checkbox if the latter one is checked (and add a tooltip explaining the dependency), or swap the conditions here if we want to minimize even if system tray icon is turned on.